packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -62,21 +62,25 @@ export class DataSourceWithBackend<
       targets = targets.filter(q => this.filterQuery!(q));
     }
     const queries = targets.map(q => {
+      let datasourceId = this.id;
```

Comment:
nit: not a big fan of `let` but I can't come up with a better solution 😞

Reply:
:)  at least with strict checking `let` means explicitly that it can change